src/Discord.Net.Commands/SlashCommands/Attributes/Choice.cs

```diff
@@ -0,0 +1,41 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Discord.SlashCommands
+{
+    /// <summary>
+    ///     Defines the parameter as a choice.
+    /// </summary>
+    [AttributeUsage(AttributeTargets.Parameter, AllowMultiple = true)]
+    public class Choice : Attribute
+    {
+        /// <summary>
+        ///     The internal value of this choice.
+        /// </summary>
+        public string choiceStringValue;
+
+        /// <summary>
+        ///     The internal value of this choice.
+        /// </summary>
+        public int? choiceIntValue = null;
+
+        /// <summary>
+        ///     The display value of this choice.
+        /// </summary>
+        public string choiceName;
```

Comment:
Heads up: Public members should be [pascal case by convention](https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/capitalization-conventions). Also, not sure if the internal values (`choiceStringValue`, `choiceIntValue`) should be public.

```suggestion
        /// <summary>
        ///     The internal value of this choice.
        /// </summary>
        private string _choiceStringValue;

        /// <summary>
        ///     The internal value of this choice.
        /// </summary>
        private int? _choiceIntValue = null;

        /// <summary>
        ///     The display value of this choice.
        /// </summary>
        public string ChoiceName { get; set; }
```